src/ray/gcs/gcs_server/gcs_actor_manager.h

```diff
@@ -462,6 +465,15 @@ class GcsActorManager : public rpc::ActorInfoHandler {
   void CancelActorInScheduling(const std::shared_ptr<GcsActor> &actor,
                                const TaskID &task_id);
 
+  /// Get the alive or dead actor of the actor id.
+  /// NOTE: The return value is not meant to be passed to other scope.
+  /// This return value should be used only for a short-time usage.
+  ///
+  /// \param actor_id The id of the actor.
+  /// \return Actor instance. The nullptr if the actor doesn't exist.
+  ///
+  const GcsActor *GetActor(const ActorID &actor_id) const;
```

Comment:
Why not use absl::optional?

Reply:
The actor information is stored as a pointer (shared_ptr<GcsActor> in the data structure), so I thought raw pointer was fine. 

Alternatively, I can just return shared pointer directly.